ouroboros-consensus/test-storage/Test/Ouroboros/Storage/VolatileDB/TestBlock.hs

```diff
@@ -84,7 +84,7 @@ parseImpl hasFS@HasFS{..} _err path =
                -> Set BlockId
                -> m ([(SlotOffset, (BlockSize, BlockInfo BlockId))], Maybe (ParserError BlockId))
             go mp n trials bids = do
-                bs <- hGet hndl binarySize
+                bs <- BL.toStrict <$> hGetLenient hasFS hndl binarySize
```

Comment:
So here you use `hGetLenient`. Can you tell me again why we can't just use `hGetExactly` here?

Reply:
Here, the parser reads blocks of known size from a file, until these blocks are over. When the blocks are over, `hGetLenient` will return a null byteString. `hGetExactly` would have thrown an exception. I think the first way is easier o deal with, than catching exceptions.